src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -603,25 +713,37 @@ protected override LocalState AllBitsSet()
             return new LocalState(BitVector.Create(nextVariableSlot), BitVector.Create(nextVariableSlot));
         }
 
-        private void EnterParameters(ImmutableArray<ParameterSymbol> parameters)
+        private void EnterParameters()
         {
-            // label out parameters as not assigned.
-            foreach (var parameter in parameters)
+            var methodParameters = ((MethodSymbol)_member).Parameters;
+            var signatureParameters = (_methodSignatureUse & MethodSignatureUse.ParameterTypes) == 0 ? default : _methodSignatureOpt.Parameters;
```

Comment:
>default [](start = 103, length = 7)

Based on the comment on `_methodSignatureUse`, I would have expected 
`var signatureParameters = (_methodSignatureUse & MethodSignatureUse.ParameterTypes) == 0 ? ((MethodSymbol)_member).Parameters: _methodSignatureOpt.Parameters;`
here.

Then we can dispense with the `IsDefault` checking below.